dart/simulation/World.cpp

```diff
@@ -74,11 +76,49 @@ World::~World()
   delete mConstraintSolver;
   delete mRecording;
 
-  for (std::vector<dynamics::Skeleton*>::const_iterator it = mSkeletons.begin();
-       it != mSkeletons.end(); ++it)
+  for(common::Connection& connection : mNameConnectionsForSkeletons)
+    connection.disconnect();
+
+  for(common::Connection& connection : mNameConnectionsForFrames)
+    connection.disconnect();
+}
+
+//==============================================================================
+WorldPtr World::clone() const
+{
+  WorldPtr worldClone(new World(mName));
+
+  worldClone->setGravity(mGravity);
+  worldClone->setTimeStep(mTimeStep);
+
+  // Clone and add each Skeleton
+  for(size_t i=0; i<mSkeletons.size(); ++i)
+  {
+    worldClone->addSkeleton(mSkeletons[i]->clone());
+  }
+
+  // Clone and add each SimpleFrame
+  for(size_t i=0; i<mFrames.size(); ++i)
+  {
+    worldClone->addFrame(mFrames[i]->clone(mFrames[i]->getParentFrame()));
+  }
+
+  // For each newly cloned SimpleFrame, try to make its parent Frame be one of
+  // the new clones if there is a match. This is meant to minimize any possible
+  // interdependencies between the kinematics of different worlds.
+  for(size_t i=0; i<worldClone->getNumFrames(); ++i)
   {
-    delete (*it);
+    dynamics::Frame* current_parent =
+        worldClone->getFrame(i)->getParentFrame();
+
+    dynamics::SimpleFramePtr parent_candidate =
+        worldClone->getFrame(current_parent->getName());
+
+    if(parent_candidate)
+      worldClone->getFrame(i)->setParentFrame(parent_candidate.get());
```

Comment:
If the parent of the SimpleFrame is another SimpleFrame which has **not** been added to the original World, then it will not be found in the new World.

The motivation behind this is you can decide which SimpleFrames are tied to their World versus which that aren't. You can have some SimpleFrames which will be copied when their World is cloned while other SimpleFrames are used across all the World copies.